docs/tutorials/migrate-to-chiselled-rock.rst

```diff
@@ -0,0 +1,268 @@
+    => CACHED [installer 2/2] RUN dotnet_version=6.0.16&& curl -fSL --output dotnet.tar.gz https://dotnet  0.0s
+    => CACHED [stage-1 2/3] COPY --from=installer [/dotnet, /usr/share/dotnet]   0.0s
+    => CACHED [stage-1 3/3] RUN ln -s /usr/share/dotnet/dotnet /usr/bin/dotnet   0.0s
+    => exporting to image   0.0s
+    => => exporting layers  0.0s
+    => => writing image sha256:a24cab51d4d02019dafcd22a2e2a3e1e6d033f9bbf1cb401d465cb2426bb2264 0.0s
+    => => naming to docker.io/library/dotnet-runtime:reference   0.0s
+
+Now, inspect this .NET reference image's size and make sure it is functional:
+
+.. literalinclude:: code/migrate-to-chiselled-rock/task.yaml
+    :language: bash
+    :start-after: [docs:inspect-docker-image]
+    :end-before: [docs:inspect-docker-image-end]
+    :dedent: 2
+
+The output should look as follows:
+
+..  code-block:: text
+    :emphasize-lines: 2
+
+    REPOSITORY TAG IMAGE ID CREATED SIZE
+    dotnet-runtime   reference   a24cab51d4d0   4 minutes ago   187MB
+
+    global.json file:
+    Not found
+
+    Host:
+    Version:6.0.16
+    Architecture: x64
+    Commit: 1e620a42e7
+
+    .NET SDKs installed:
+    No SDKs were found.
+
+    .NET runtimes installed:
+    Microsoft.NETCore.App 6.0.16 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
+
+    Download .NET:
+    https://aka.ms/dotnet-download
+
+    Learn about .NET Runtimes and SDKs:
+    https://aka.ms/dotnet/runtimes-sdk-info
+
+
+Convert Dockerfile to rockcraft.yaml file
+-----------------------------------------
+
+From a quick analysis of the reference Dockerfile above, the following
+requirements must be met:
+
+- R1. the ROCK must be based on Ubuntu Jammy
+- R2. there is no predefined Entrypoint or default command
+- R3. The ROCK must have version 6.0 of the .NET Runtime installed
+- R4. ``/usr/bin/dotnet`` must be a symbolic link to the .NET binary
+
+With these requirements in mind, and in the same directory as the Dockerfile
+from above, write the following into a text editor and save it as
+``rockcraft.yaml``:
+
+.. literalinclude:: code/migrate-to-chiselled-rock/rockcraft.yaml
+    :language: yaml
+
+Note the subtle chiselling of the .NET Runtime package in the ``rockcraft.yaml``
+file above. You are requesting Rockcraft to install the ``libs`` slice of the
+``dotnet-runtime`` deb, which is defined in the
+`ubuntu-22.04 Chisel release
+<https://github.com/canonical/chisel-releases/blob/ubuntu-22.04/slices/>`_.
+
+
+Pack the Chiselled ROCK with Rockcraft
+--------------------------------------
+
+To build the ROCK, run:
+
+.. literalinclude:: code/migrate-to-chiselled-rock/task.yaml
+    :language: bash
+    :start-after: [docs:build-rock]
+    :end-before: [docs:build-rock-end]
+    :dedent: 2
+
+The output should be similar to:
+
+..  code-block:: text
+    :emphasize-lines: 42
+
+    2023-04-19 15:52:48.045 Starting Rockcraft 0.0.1.dev1
```

Comment:
I wonder if this verbose output is useful? Isn't it too overwhelming for new users?